cosmoz-default-tree.html

```diff
@@ -15,142 +27,236 @@
+	 * @param {object} nodes [this._roots] (The nodes the search should be based on.)
+	*/
+	Cosmoz.DefaultTree.prototype._searchNodes = function (propertyValue, options, nodes) {
+		var results = [];
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		nodes = nodes || this._roots;
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
-	};
+		nodes.some(function (node) {
+			results = results.concat(this.search(node, propertyValue, options));
+			return options.firstHitOnly && results.length > 0;
+		}, this);
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
-			return;
-		}
-		return _objectValues(node.children);
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getProperty = function (node, propertyName) {
-		if (node === undefined || propertyName === undefined) {
-			return;
+	/**
+	 * Returns the node of a given path.
+	 * @return {object}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj [this._treeData] (The object the search should be based on.)
+	 * @param {string} separatorSign ["."] (The string which separates the path. e.g ".")
+	*/
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator, nodeObj, separatorSign) {
+		if (!pathLocator) {
+			return this._roots;
 		}
 
-		return node[propertyName];
-	};
+		// Defaults
+		nodeObj = nodeObj || this._treeData;
+		separatorSign = separatorSign || '.';
 
-	Cosmoz.DefaultTree.prototype._searchAllRoots = function (propertyName, propertyValue) {
-		var i,
-			node;
+		var	pathNodes = this.getPathNodes(pathLocator, nodeObj, separatorSign);
+		return pathNodes && pathNodes.pop();
+	};
 
-		for (i = 0; i < this._roots.length; i+=1) {
-			node = this.search(this._roots[i], propertyName, propertyValue);
-			if (node !== undefined) {
-				return node;
-			}
+	/**
+	 * Returns the nodes on a given path.
+	 * @return {array}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj [this._treeData] (The object the search should be based on.)
+	 * @param {string} separatorSign ["."] (The string which separates the path.)
+	*/
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodeObj, separatorSign) {
+		if (!pathLocator) {
+			return this._roots;
 		}
-	};
 
-	Cosmoz.DefaultTree.prototype._getNodeByPathLocator = function (pathLocator) {
-		var	node,
-			path,
-			i;
+		// Defaults
+		separatorSign = separatorSign || '.';
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodeObj || this._treeData,
+			nodes = [];
 
-		for (i = 0; i < this._roots.length; i+=1) {
-			node = this._roots[i];
-			if (node.pathLocator === pathLocator) {
-				return node;
+		// Get the nodes on the path.
+		path.some(function (nodeKey) {
```

Comment:
Well the case you're missing is the one I referred to as #2 - when there are missing parts in the beginning of the path.
`return nodes.length === 0;` will validate missing node parts as long as they're in the beginning of the array. That creates the corner case that all node parts are missing but path is still validated.
But if it's hard to understand the readability is too bad.